### PR TITLE
Use correct README link to render the README

### LINKED ISSUE #23151: Repository homepage readme cannot render images of the repository.
### Description

_No response_

### Gitea Version

v1.19+

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

image
image

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

./gitea web

### Database

None

### PR DESCRIPTION
**Pull request by yp05327**
`renderReadmeFile` needs `readmeTreelink` as parameter but gets `treeLink`.
The values of them look like as following:
`treeLink`:  `/{OwnerName}/{RepoName}/src/branch/{BranchName}`
`readmeTreelink`: `/{OwnerName}/{RepoName}/src/branch/{BranchName}/{ReadmeFileName}`

`path.Dir` in
https://github.com/go-gitea/gitea/blob/8540fc45b11eff9a73753ca139f8ea5c38509bf5/routers/web/repo/view.go#L316
should convert  `readmeTreelink` into `/{OwnerName}/{RepoName}/src/branch/{BranchName}` instead of the current `/{OwnerName}/{RepoName}/src/branch`.

Fixes #23151

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Comment by lunny** [comment]
> `renderReadmeFile` needs `readmeTreelink` as parameter but gets `treeLink`. The values of them look like as following: `treeLink`: `/{OwnerName}/{RepoName}/src/branch/{BranchName}` `readmeTreelink`: `/{OwnerName}/{RepoName}/src/branch/{BranchName}/{ReadmeFileName}`
> 
> `path.Dir` in
> 
> https://github.com/go-gitea/gitea/blob/8540fc45b11eff9a73753ca139f8ea5c38509bf5/routers/web/repo/view.go#L316
> 
> should convert `readmeTreelink` into `/{OwnerName}/{RepoName}/src/branch/{BranchName}` instead of the current `/{OwnerName}/{RepoName}/src/branch`.
> 
> Fixes #23151

Could you add some tests for that?

**Review by yardenshoham — APPROVED** [review]

**Comment by lunny** [comment]
Maybe you can merge #23246 into this PR to ensure tests PASS?

**Comment by yp05327** [comment]
> Maybe you can merge #23246 into this PR to ensure tests PASS?

Done.

File: tests/integration/repo_test.go [hunk 1]
```diff
@@ -256,3 +256,23 @@ func TestViewRepoDirectory(t *testing.T) {
...
+func TestMarkDownImage(t *testing.T) {
+	defer tests.PrepareTestEnv(t)()
+
+	session := loginUser(t, "user2")
+
+	req := NewRequest(t, "GET", "/user2/repo1/src/branch/home-md-img-check")
+	resp := session.MakeRequest(t, req, http.StatusOK)
+
+	htmlDoc := NewHTMLParser(t, resp.Body)
+	_, exists := htmlDoc.doc.Find("img[src=\"/user2/repo1/media/branch/home-md-img-check/test-fake-img.jpg\"]").Attr("src")
```
Comment (silverwind) [inline tests/integration/repo_test.go]:
```suggestion
	_, exists := htmlDoc.doc.Find(`img[src="/user2/repo1/media/branch/home-md-img-check/test-fake-img.jpg"]`).Attr("src")
```

File: tests/integration/repo_test.go [hunk 2]
```diff
@@ -256,3 +256,23 @@ func TestViewRepoDirectory(t *testing.T) {
...
+	assert.True(t, exists, "Repo home page markdown image link check failed")
+
+	req = NewRequest(t, "GET", "/user2/repo1/src/branch/home-md-img-check/README.md")
+	resp = session.MakeRequest(t, req, http.StatusOK)
+
+	htmlDoc = NewHTMLParser(t, resp.Body)
+	_, exists = htmlDoc.doc.Find("img[src=\"/user2/repo1/media/branch/home-md-img-check/test-fake-img.jpg\"]").Attr("src")
```
Comment (silverwind) [inline tests/integration/repo_test.go]:
```suggestion
	_, exists = htmlDoc.doc.Find(`img[src="/user2/repo1/media/branch/home-md-img-check/test-fake-img.jpg"]`).Attr("src")
```

File: tests/integration/repo_test.go (hunk 1 above)
Comment (yp05327) [reply tests/integration/repo_test.go]: In some places, we are using the same format.
Such as: 
https://github.com/go-gitea/gitea/blob/ffce336f1802b2f2298fc8fd27e815086702c812/tests/integration/repo_generate_test.go#L39
Is it necessary to fix them all in another PR?

**Comment by kousu** [comment]
Sorry about that. Thank you for fixing my bug @yp05327

**Pull request merged**